File: hackerrank/easy/0017-electronics-shop/swift/solution.swift
// solution.swift

func getMoneySpent(keyboards: [Int], drives: [Int], b: Int) -> Int {
    let ks = keyboards.sorted()
    let ds = drives.sorted(by: >)
    var i = 0
    var j = 0
    var best = -1

    while i < ks.count && j < ds.count {
        let s = ks[i] + ds[j]
        if s > b {
            j += 1    // too expensive, try cheaper drive
        } else {
            if s > best { best = s }
            i += 1    // try a pricier keyboard
        }
    }
    return best
}

// --- demo run ---
print(getMoneySpent(keyboards: [3, 1], drives: [5, 2, 8], b: 10))   // 9
print(getMoneySpent(keyboards: [4], drives: [5], b: 5))   // -1

File: hackerrank/easy/0017-electronics-shop/python/solution.py
## solution.py

from typing import List

def get_money_spent(keyboards: List[int], drives: List[int], b: int) -> int:
    ks = sorted(keyboards)
    ds = sorted(drives, reverse=True)
    i = j = 0
    best = -1

    while i < len(ks) and j < len(ds):
        s = ks[i] + ds[j]
        if s > b:
            j += 1    # try cheaper drive
        else:
            if s > best:
                best = s
            i += 1    # try pricier keyboard
    return best


# --- demo run ---
if __name__ == "__main__":
    print(get_money_spent([3, 1], [5, 2, 8], 10))     # 9
    print(get_money_spent([4], [5], 5))   # -1